### PR TITLE
LibWeb/CSS: Implement add for FontFaceSet

### PR DESCRIPTION
**Pull request by kostyafarber**
There is still some work to do with some of the underying methods called inside this method (e.g is_css_connected) but this is a start.

### CODE REVIEW

**Review by AtkinsSJ — APPROVED** [review]
Welcome! All looks correct to me. :^)

Having a test too would be nice - you could do something similar to https://wpt.live/css/css-font-loading/fontfaceset-add-css-connected.html maybe. I'm not sure if enough of the other `FontFaceSet` code is implemented to spec to make that pass though. If not, just say so.

**Comment by kostyafarber** [comment]
Thanks for the quick response time to review! Nice, I'll take a look at and see if I can whip something up. Either way I'll ping here once I've got the commit or lack thereof if it's not possible. Thanks.

File: Userland/Libraries/LibWeb/CSS/FontFaceSet.h [hunk 1]
```diff
@@ -50,7 +50,11 @@ class FontFaceSet final : public DOM::EventTarget {
     virtual void visit_edges(Cell::Visitor&) override;
 
     JS::NonnullGCPtr<JS::Set> m_set_entries;
-    JS::GCPtr<WebIDL::Promise> m_ready_promise; // [[ReadyPromise]]
+    JS::GCPtr<WebIDL::Promise> m_ready_promise;      // [[ReadyPromise]]
+                                                     //
+    Vector<JS::Handle<FontFace>> m_loading_fonts {}; // [[LoadingFonts]]
+    Vector<JS::Handle<FontFace>> m_loaded_fonts {};  // [[LoadedFonts]]
+    Vector<JS::Handle<FontFace>> m_failed_fonts {};  // [[FailedFonts]]
```
Comment (ADKaster) [inline Userland/Libraries/LibWeb/CSS/FontFaceSet.h:57]: I'm skeptical that we need to hold onto a vector of handles. if we visit every element in each list in visit_edges, it should be fine to have a `Vector<JS::NonnullGCPtr<FontFace>>`

Comment (kostyafarber) [reply Userland/Libraries/LibWeb/CSS/FontFaceSet.h:57]: Okay I'll make that change. Thanks.

**Comment by AtkinsSJ** [comment]
Please drop the commits you've picked up by Andreas and squash your changes into the first commit. We like corrections to a PR to go into the existing commits as it makes the commit history easier to follow.

**Comment by kostyafarber** [comment]
Sure thing, I wasn't sure if we rebase/merge our PRs but went ahead with it, I'll clean it up so all my changes are in the first commit!

File: Userland/Libraries/LibWeb/CSS/FontFaceSet.h [hunk 2]
```diff
@@ -50,8 +50,12 @@ class FontFaceSet final : public DOM::EventTarget {
...
 
     JS::NonnullGCPtr<JS::Set> m_set_entries;
-    JS::GCPtr<WebIDL::Promise> m_ready_promise; // [[ReadyPromise]]
-
+    JS::GCPtr<WebIDL::Promise> m_ready_promise;            // [[ReadyPromise]]
+                                                           //
+    Vector<JS::NonnullGCPtr<FontFace>> m_loading_fonts {}; // [[LoadingFonts]]
+    Vector<JS::NonnullGCPtr<FontFace>> m_loaded_fonts {};  // [[LoadedFonts]]
+    Vector<JS::NonnullGCPtr<FontFace>> m_failed_fonts {};  // [[FailedFonts]]
+                                                           //
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/FontFaceSet.h:58]: You need to visit() these in `visit_edges()`, which is how the GC finds what's being referenced. Should just be a case of `visitor.visit(m_loading_fonts)` etc.

Also this trailing line with a blank comment is a bit silly. :^)

Comment (kostyafarber) [reply Userland/Libraries/LibWeb/CSS/FontFaceSet.h]: Ah okay, sorry still trying to get a solid understanding of the required actions when doing these implementations.

Agreed! my silly editor inserts them when there is a comment above the line I'm editing and I missed it. I'll remove.

Comment (kostyafarber) [reply Userland/Libraries/LibWeb/CSS/FontFaceSet.h]: done :)

**Review by ADKaster — APPROVED** [review]

**Pull request merged**